### ISSUE TITLE
Sound before and after say and ssml

### ISSUE THREAD

**Issue by realwax** [Apollon77/ioBroker.alexa2#657]
Since yesterday my say or ssml commands are surrounded by a sinus tone in the beginning and end. Pretty annoying.  It was not there until yet, haven't changed anything. Is there a way to turn it off? Haven't found a setting in the adapter or object nor a feature in the app. It's not the notification. Thank you!

**Comment by Apollon77** [comment]
Honestly ... what do you think it most likely is when you did not updated the adapter? Right ... a change on amazon side ... can you reproduce that too when you use a routine via the app to do such a speak? if yes report it to amazon that your routine is giving that.
Wait some day if it solves from itself by amazon seeing there error and rolling it back

**Comment by realwax** [comment]
I never meant that it is somebodies fault. I just noticed an older notification issue/question and though maybe it's related.
Sure Amazon is messing around. I noticed that a lot since I am using language tags in ssml and say to have english words pronounced correctly and not in German. Every third day I am getting the "there are problems with the simonsays skill" message. Not related to that but seems they are devOps-ing JIT=PROD until it's dead :D 

If you meant the routine in the app - tiggered by a phrase and letting Echo reply - that still works without the tone... 

So probably a change on the API side, or the way how say calls are being processed. Maybe...

**Comment by realwax** [comment]
Nevertheless - trying to a raise a ticket at AMZN.... thanks for your quick reply!

**Comment by realwax** [comment]
In operation with Amazon support - device will factory reseted. Confirmation of a firmware rollout yesterday. As rollout is phased not everybody will be aware.... Stay tuned.

**Comment by skillix** [comment]
209 / 5000
Résultats de traduction
hello, I opened an anomaly at amzon france, it is bug of version 4799221636 echo dot.
the sound comes from the change in volume. try to press the physical volume button, the sound is the same

**Comment by padrino-x** [comment]
@skillix From the firmware, it seems like your device is 3rd gen or newer?
@realwax Are your devices, too?
I only have 2nd gen devices here and they only have that sound when physically pressing +/- button, but not when triggering through the adapter/app. 🤞

**Comment by realwax** [comment]
Yes, @padrino-x 3rd gen.

**Comment by skillix** [comment]
@padrino-x  Yes I only have 3rd generation, I have 2 with this firmware which have noise and 3 with the version before which does not. this is what allowed me that the problem came from this version. However, I have not had any information on the problem with the 4th generation, so I don't know if the problem is also present in the 4th generation.

**Comment by realwax** [comment]
As not every x-gen device is affected it seems to be firmware related or a backend change and translation only affecting 3rd and maybe newer. Maybe it makes sense to double check API changes. I am puzzled why other skills don't suffer from this....

**Comment by skillix** [comment]
the problem is rated amazon, the API is not the cause, the bug can be duplicated with a simple routine via alexa app

**Comment by realwax** [comment]
> the problem is rated amazon, the API is not the cause, the bug can be duplicated with a simple routine via alexa app

Very interesting. I can't replicate the issue with a custom routine set up with the Alexa app. Let's I create a dummy like "I say "test" and let Alexs reply a certain sentence I defined in the app" I this case I do not get the tones before and after. 
I only get it when using iobroker.iot with say and ssml

**Comment by skillix** [comment]
> > the problem is rated amazon, the API is not the cause, the bug can be duplicated with a simple routine via alexa app
> 
> Very interesting. I can't replicate the issue with a custom routine set up with the Alexa app. Let's I create a dummy like "I say "test" and let Alexs reply a certain sentence I defined in the app" I this case I do not get the tones before and after.
> I only get it when using iobroker.iot with say and ssml

102 / 5000
Résultats de traduction
small precision, the sound appears only if there is a change of volume, via API or via Routine alexa

**Comment by skillix** [comment]
in reality there should be 2 sound change functions, one in silence for the routine, and one for the physical button. and in this version they merge the 2

**Comment by padrino-x** [comment]
@skillix 
Of course, I don't know Amazon's intentions, but it could be to support handicaped/blind people to give them audio feedback.
It used to be that the Echos here made the sound too, when I asked by voice for the volume to change.
This is gone now (I wouldn't mind to have it in that case, gives me no need to look at the device, still knowing I was understood).
I agree with you, there should be two cases, 1) automated changes (ie. scripts, routines...) and direct user input (wether physical or, if they like, by voice).
The problem is, it's Amazon to choose.
And it is unlikely they will listen to user input - never seen them do it (at least not for my inputs :-D)

**Comment by realwax** [comment]
I am still in touch with amazon. The problem is that since the fw update volume changes via media.tts produce the same signal sound as when the hardware button is pressed. Looking into the iobroker alexa code it is obvious the volume is set and therefore we have the tone....

**Comment by realwax** [comment]
Still no update from Amazon. Error is simply reproducable with setting the volume in the app. A look into the code shows that volume is set every say or ssml which seems desired as it would not be a problem when device does work as used to be. On gen 4 devices this error seems not to be there.

**Comment by padrino-x** [comment]
Maybe there us hope for amazon changing it back, as it does not happen with 4th and 2nd generation devices...

**Comment by realwax** [comment]
They will! ;) Otherwise they need an extra employee for my weekly call :P As I questioned this behavior as desired the support agent  tested and verified the not affected 4th generation. Therefore a bug and me in relentless mode.... :D

**Comment by skillix** [comment]
hello, my 2 echo with the old update were upgrade in 4799221636, but it does not produce the sound. I guess a fix was made to this version but not propagated on the already installed echo.
hope for the next one

**Comment by realwax** [comment]
@skillix Thanks for the info! I share the same version number but obviously without the fix. (This is crazy if they fixed it without changing the version number...)

**Comment by LosFoppos** [comment]
same here...on 4799221636. Is itpossible to downgrade the echo?

**Comment by skillix** [comment]
To reset your device to its factory settings:

1 - Press and hold the Action button for 25 seconds. The light ring will pulse orange, then turn off.
2 - Wait for the light ring to turn back on and turn blue. The light ring then turns orange again and the device enters the set up mode.

I don't know if this changes the version, I have too many home automation scenarios to test

**Comment by padrino-x** [comment]
I don't think think a downgrade is possible. AFAIK the firmware stays.
And even if it would be in delivery state version, the bugs in such an old firmware are probably many more. 😅

**Comment by realwax** [comment]
I did try a factory reset as part of the support routine. It did not change a thing. Routines where still there after the reset as they are saved in the cloud and seem not to be deleted. So if they really fixed it within the same version only the next firmware will show. I haven't received an answer yet from Amazon.
